File: scripts/check-jobs-status.py
#!/usr/bin/env python3
"""Verificar status das coletas de vagas"""
import psycopg2
import os
from dotenv import load_dotenv
from datetime import datetime

load_dotenv()

conn = psycopg2.connect(
    host=os.getenv('POSTGRES_HOST'),
    port=os.getenv('POSTGRES_PORT', '5432'),
    user=os.getenv('POSTGRES_USER'),
    password=os.getenv('POSTGRES_PASSWORD'),
    database=os.getenv('POSTGRES_DB')
)

cur = conn.cursor()

print("=" * 70)
print("STATUS DAS COLETAS DE VAGAS")
print("=" * 70)

# Total e datas
cur.execute("SELECT COUNT(*), MAX(collected_at), MIN(collected_at) FROM sofia.jobs")
total, last, first = cur.fetchone()
print(f"\n📊 RESUMO GERAL:")
print(f"   Total de vagas: {total}")
print(f"   Primeira coleta: {first}")
print(f"   Última coleta: {last}")

if last:
    days_ago = (datetime.now() - last.replace(tzinfo=None)).days
    print(f"   ⏰ Última coleta foi há {days_ago} dias")

# Por plataforma
print(f"\n📱 POR PLATAFORMA:")
cur.execute("""
    SELECT platform, COUNT(*), MAX(collected_at), MIN(collected_at)
    FROM sofia.jobs 
    GROUP BY platform 
    ORDER BY MAX(collected_at) DESC NULLS LAST
""")

for platform, count, last_col, first_col in cur.fetchall():
    if last_col:
        days = (datetime.now() - last_col.replace(tzinfo=None)).days
        status = "🟢" if days < 7 else ("🟡" if days < 30 else "🔴")
        print(f"   {status} {platform}: {count} vagas (última: {last_col.strftime('%Y-%m-%d')} - há {days} dias)")
    else:
        print(f"   ⚪ {platform}: {count} vagas (sem data de coleta)")

# Vagas recentes (últimos 30 dias)
print(f"\n📅 VAGAS DOS ÚLTIMOS 30 DIAS:")
cur.execute("""
    SELECT platform, COUNT(*) 
    FROM sofia.jobs 
    WHERE collected_at >= NOW() - INTERVAL '30 days'
    GROUP BY platform
    ORDER BY COUNT(*) DESC
""")
recent = cur.fetchall()
if recent:
    for platform, count in recent:
        print(f"   - {platform}: {count} vagas")
else:
    print("   ❌ NENHUMA vaga coletada nos últimos 30 dias!")

conn.close()
print("=" * 70)

File: scripts/check-jobs-status.ts
import { Pool } from 'pg';
import * as dotenv from 'dotenv';
dotenv.config();

const pool = new Pool({
    host: process.env.POSTGRES_HOST || 'localhost',
    port: parseInt(process.env.POSTGRES_PORT || '5432'),
    user: process.env.POSTGRES_USER || 'sofia',
    password: process.env.POSTGRES_PASSWORD || 'sofia123strong',
    database: process.env.POSTGRES_DB || 'sofia_db',
});

async function main() {
    console.log('🔍 Verifying Job Collection Status...');

    try {
        // Check sofia.jobs
        const jobsCount = await pool.query(`
            SELECT COUNT(*) as count 
            FROM sofia.jobs 
            WHERE collected_at >= CURRENT_DATE;
        `);
        console.log(`💼 'sofia.jobs' collected today: ${jobsCount.rows[0].count}`);

        // Check sofia.tech_jobs (if distinct)
        const techJobsCount = await pool.query(`
            SELECT COUNT(*) as count 
            FROM sofia.tech_jobs 
            WHERE collected_at >= CURRENT_DATE;
        `);
        console.log(`💻 'sofia.tech_jobs' collected today: ${techJobsCount.rows[0].count}`);

        // Check source breakdown for today
        const sourceBreakdown = await pool.query(`
            SELECT source, COUNT(*) as count
            FROM sofia.jobs
            WHERE collected_at >= CURRENT_DATE
            GROUP BY source
            ORDER BY count DESC;
        `);

        console.log('\n📊 Breakdown by Source (Today):');
        if (sourceBreakdown.rows.length === 0) {
            console.log('   (No jobs collected today)');
        } else {
            sourceBreakdown.rows.forEach(r => {
                console.log(`   • ${r.source}: ${r.count}`);
            });
        }

    } catch (err) {
        console.error('❌ Error checking jobs:', err.message);
    } finally {
        await pool.end();
    }
}

main();
